infrastructure_source_code.md

```diff
@@ -0,0 +1,44 @@
+# DataWorks Open Source Infrastructure Source Code Policy
+
+## Purpose
+
+This document sets out local policy, for the Department for Work and Pensions (DWP) DataWorks team, on what types of information can and cannot be included in infrastructure code that is made publicly available.
+
+It is expected that, at some point, this document will be superceded by a DWP-wide open-source infrastructure code policy which encompasses all DWP public GitHub repositories. Some useful [DWP guidance on creating open source products](https://confluence.service.dwpcloud.uk/pages/viewpage.action?title=Guidance+on+creating+open+source+products&spaceKey=EN) currently exists, but it has not yet been made open-source.
+
+## Scope
+
+This policy covers infrastructure code. Please refer to the [index](README.md#index) for links to other related polices.
+
+## General Approach
+
+Infrastructure code should be made open wherever it is appropriate to do so:
+
+This approach makes code more readily accessible to a broad target audience, so that they can be more easily used for their intended purpose;
+1. This approach enables and encourages re-use of common standards across the DWP and other Government departments, which in turn makes collaboration easier. It also enables other parties to feed back improvements, which benefits everyone.
+1. This policy is in line with the spirit of the  [Government Digital Service (GDS) guidelines on making information open-source](https://gds-operations.github.io/guidelines/).
+
+## Security Restrictions
+
+Reference information which could be used to support targeted attacks on DataWorks or other DWP resources must not be published in open repositories.
+
+Sensitive reference information can take several forms. A non-exhaustive list of these is provided below:
```

Comment:
We need to consider things such as role names. These provide information to a potential attacker which can be directly used in speculative attacks against the environment eg brute-forcing attacks etc. When used with e.g. lists of compromised user credentials from other services, some of whom might be reasonably guessed to be working on our platform, that could become a powerful tool for an attacker. We don't necessarily need to exclude these, but would need to consider our strategy for mitigating the risk.

Reply:
I understand that a role arn could be used this way, but not a role name

Reply:
Your argument being that if, for example, the account number is kept private then the name of the role on its own has no value to an attacker? There's merit in that approach. Perhaps a quick face-to-face discussion?